### PR TITLE
fix(Scripts/UBRS): The Beast should deaggro when walking over the fir…

### LINKED ISSUE #9127: [UBRS] The Beast Issues
### Current Behaviour
 
The beast can walk over the firewall
The beast casts fireball and then directly fireblast always
The RP doesn't always work, with the two guards not running away and dieing, the beast does come out usually.

### Expected Blizzlike Behaviour

The beast should deaggro when walking over the firepit walls and reset.
The beast should not use fireball and then blastwave at the same time, this should be more random
The RP should work, where the two guards near the beast should run away, and then get killed by him, then he should walk around the room (only last part works correctly 100% of the time at the moment)

### Source

https://www.wowhead.com/npc=10430/the-beast#comments

### Steps to reproduce the problem

Go to UBRS
Walk in the room before the beast at normal speed to observe
dont immidiately teleport to the beast.

### AC rev. hash/commit

06bc64d1c370

### Operating system

Win64

### Custom changes or Modules

Eluna

### PR DESCRIPTION
**Pull request by UltraNix**
…epit walls.

The Beast should not use fireball and then fireblast at the same time.
Fixes #9127

## Issues Addressed:

- Closes #9127

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

Go to UBRS
Walk in the room before the beast at normal speed to observe
dont immidiately teleport to the beast.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by temperrr** [comment]
Tested:

- [ ]teleported to .tele ubrs  I walked in slowly to the beast, I could still move him over the firewall.
image

**Comment by savowe** [comment]
The beast would not follow me behind the firewall

https://user-images.githubusercontent.com/38558793/142076115-eb8debd1-e892-44f1-bda8-900451602344.mp4

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_the_beast.cpp [hunk 1]
```diff
@@ -217,10 +226,14 @@ class boss_the_beast : public CreatureScript
                     case EVENT_FIREBALL:
                         DoCastVictim(SPELL_FIREBALL);
                         events.ScheduleEvent(EVENT_FIREBALL, 8 * IN_MILLISECONDS, 21 * IN_MILLISECONDS);
+                        if (events.GetNextEventTime(EVENT_FIREBLAST) < 3 * IN_MILLISECONDS)
+                            events.RescheduleEvent(EVENT_FIREBLAST, 3 * IN_MILLISECONDS);
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_the_beast.cpp:230]: brackets

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_the_beast.cpp [hunk 2]
```diff
@@ -217,10 +226,14 @@ class boss_the_beast : public CreatureScript
...
                         break;
                     case EVENT_FIREBLAST:
                         DoCastVictim(SPELL_FIREBLAST);
                         events.ScheduleEvent(EVENT_FIREBLAST, 5 * IN_MILLISECONDS, 8 * IN_MILLISECONDS);
+                        if (events.GetNextEventTime(EVENT_FIREBALL) < 3 * IN_MILLISECONDS)
+                            events.RescheduleEvent(EVENT_FIREBALL, 3 * IN_MILLISECONDS);
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_the_beast.cpp:236]: same

**Review by Nefertumm — APPROVED** [review]

**Pull request merged**